packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -349,15 +370,48 @@ export class ForkChoice implements IForkChoice {
       this.updateJustified(currentJustifiedCheckpoint, justifiedBalances);
     }
 
-    const targetSlot = computeStartSlotAtEpoch(computeEpochAtSlot(slot));
     const blockRoot = this.config.getForkTypes(slot).BeaconBlock.hashTreeRoot(block);
+    const blockRootHex = toHexString(blockRoot);
+
+    if (this.proposerBoostEnabled && slot === this.fcStore.currentSlot) {
+      let secondsIntoSlot;
+      if (preCachedData?.blockReceptionTime !== undefined) {
```

Comment:
why do we have this `blockReceptionTime` as optional? we should always have it

Reply:
reason for this was the way spec tests steps are , update time, block, ... , so if blockReceptionTime is not provided it uses time from the forkchoice secondsIntoSlot, however it can be made mandatory and then i can pull time using getSecondsIntoSlot and pass it as blockReceptionTime in spec

Reply:
Lighthouse implementation looks so simple! Is the complexity here necessary? https://github.com/sigp/lighthouse/blob/a836e180f9ad51f31767c5ffb33bebdeff1a9f3f/consensus/fork_choice/src/fork_choice.rs#L549